src/Idea.hs

```diff
@@ -79,22 +78,28 @@ showEx tt Idea{..} = unlines $
     f "Found" (Just ideaFrom') ++ f "Perhaps" ideaTo' ++
     ["Note: " ++ n | let n = showNotes ideaNote, n /= ""]
     where
-        (ideaFrom', ideaTo') = maybe (ideaFrom, Nothing) (second Just.(reduceJunk ideaFrom)) ideaTo
+        (ideaFrom', ideaTo') = maybe (ideaFrom, Nothing) ((second Just) . reduceJunk . (ideaFrom, )) ideaTo
         f msg Nothing = []
         f msg (Just x) | null xs = [msg ++ " you should remove it."]
                        | otherwise = (msg ++ ":") : map ("  "++) xs
             where xs = lines $ tt x
 
-reduceJunk :: String -> String -> (String, String)
-reduceJunk from to =
-    let
-        (tup, rev_tup) = (***) (both lines) (both (reverse.lines)) (dupe (from, to))
+-- traverse through hints output from both direction to take out all the excess 
+reduceJunk :: (String, String) -> (String, String)
+reduceJunk x@(from, to) =
+    let 
+        tup = both lines x; rev_tup = both (reverse. lines) x
+        -- compare each line to get the first dissimilar position
         go :: ([String], [String]) -> Int 
         go ([], _) = 0
         go (_, []) = 0
         go (f : fs, t : ts) = if f == t then 1 + go (fs, ts) else 0
-        (start, end) = ( 1 `max` go tup - 1, 1 `max` go rev_tup)
-    in both (unlines.(\xs-> (xs !!) <$> [start..(length xs - end)])) tup
+        -- subtract 1 in both version to get the corresponding above/below line for context.
+        (start, end) = (go tup - 1, go rev_tup - 1)      
+        -- extract a sub-array from l to r where r is the index in reversed version
+        extract :: [String] -> Int -> Int -> [String]
+        extract hints l r = drop l $ take (length hints - r) hints
```

Comment:
It's possible to avoid using `length hints - r`, please try using `l` and `r` only. Then, you can also make this function point-free.

Reply:
```haskell
extract = (. drop) . (.) . dropSuffix
```
I can change the function to this with an extra `dropSuffix` function, but I'm not sure whether it's more intelligible

Reply:
How's about this?
```haskell
extract l r = drop l . take r
```

Reply:
Sorry for the unclear code, but the extract function here is actually removing the first `l` elements and the last `r` elements. Should I change it to get rid of the confusion?